examples/src/fofb_processing.c

```diff
@@ -1,272 +1,298 @@
-          "  -b  --brokerendp <Broker endpoint>            Broker endpoint\n"
-          "  -v  --verbose                                 Verbose output\n"
-          "  -o  --boardslot <Board slot number = [1-12]>  Board slot number\n"
-          "  -s  --halcsnumber <HALCS number = [0|1]>      HALCS number\n"
-          "  -w  --ram_write_en <[Read = 0 | Write = 1]>   RAM write enable\n"
-          "  -d  --ram_data_in                             RAM data\n"
-          "  -a  --ram_addr                                RAM addr\n",
-          program_name);
-}
+static const char *short_opts = "ho:n:c:gs:b:v";
+
+struct optional_args {
+  char *broker_endp;
+  int verbose;
+};
 
-/* Our read function */
-typedef halcs_client_err_e (*halcs_func_fp) (halcs_client_t *self, char *service,
-        uint32_t chan, uint32_t *value);
+enum operation {
+  SET_COEFFS,
+  GET_COEFFS
+};
+
+enum necessary_args_idx {
+  BOARD_SLOT,
+  HALCS_NUMBER,
+  CHANNEL,
+  OP
+};
+
+#define NUM_OF_NECESSARY_ARGS 4
+struct necessary_args {
+  bool is_given[NUM_OF_NECESSARY_ARGS];
 
-typedef struct {
-  halcs_func_fp func;
-  const char *func_name;
-  int call;
-} func_call_t;
+  uint32_t board_slot;
+  uint32_t halcs_number;
+  uint32_t channel;
+  enum operation op;
+};
+
+void print_usage(char *program_name) {
+  fprintf(stdout,
+    "HALCSD FOFB Processing Client Example\n"
+    "Usages:\n"
+    "       %s --help\n"
+    "       %s [OPTIONALS] --board-slot <1-12> --halcs-number <0|1> --channel <0-11> --get-coeffs\n"
+    "       %s [OPTIONALS] --board-slot <1-12> --halcs-number <0|1> --channel <0-11> --set-coeffs <filename>\n"
+    "\n"
+    "  -h  --help                                    display this usage info\n"
+    "  -o  --board-slot <board slot number = [1-12]> board slot number\n"
+    "  -n  --halcs-number <HALCS number = [0|1]>     HALCS number\n"
+    "  -c  --channel <channel number = [0-11]>       channel number\n"
+    "  -g  --get-coeffs                              gets coeffs\n"
+    "  -s  --set-coeffs <filename>                   sets coeffs\n"
+    "\n"
+    "OPTIONALS:\n"
+    "  -b  --broker-endp <broker endpoint>           broker endpoint (defaults to %s)\n"
+    "  -v  --verbose                                 verbose output\n",
+    program_name, program_name, program_name, "ipc://"DFLT_BIND_FOLDER);
+}
 
-int main (int argc, char *argv [])
-{
+int main(int argc, char *argv[]) {
   int ret = 0;
-  int verbose = 1;
-  char *broker_endp = NULL;
-  char *board_number_str = NULL;
-  char *halcs_number_str = NULL;
-  char *ram_write_en_str = NULL;
-  char *ram_data_in_str = NULL;
-  char *ram_addr_str = NULL;
-  int opt;
 
-  while ((opt = getopt_long (argc, argv, shortopt, long_options, NULL)) != -1) {
-    /* Get the user selected options */
-    switch (opt) {
-      /* Display Help */
+  struct optional_args opt_args = {
+    .broker_endp = strdup("ipc://"DFLT_BIND_FOLDER),
```

Comment:
```suggestion
    .broker_endp = strdup("ipc://" DFLT_BIND_FOLDER),
```
Um teco mais fácil de ler